resotocore/resotocore/db/entitydb.py

```diff
@@ -96,8 +101,11 @@ async def update(self, t: T) -> T:
                 setattr(t, "revision", result["_rev"])
         return t
 
-    async def delete(self, key_or_object: Union[str, T]) -> None:
-        key = key_or_object if isinstance(key_or_object, str) else self.key_of(key_or_object)
+    async def delete(self, key: K) -> None:
```

Comment:
Why not leave it as `Union`?

Reply:
I decided to separate this into delete and delete_value methods since it looks a bit prettier in this case.